react/utils/pages.ts

```diff
@@ -154,29 +154,36 @@ export function navigate(
   const {
     page,
     params,
-    query,
-    to,
+    query: inputQuery,
+    to: inputTo = '',
     scrollOptions,
     fallbackToWindowLocation = true,
     rootPath,
     replace,
     fetchPage = true,
   } = options
 
-  if (!page && !to) {
+  if (!page && !inputTo) {
     console.error(
       `Invalid navigation options. You should use 'page' or 'to' parameters`
     )
     return false
   }
+  
+  const [to, extractedQuery] = inputTo.split('?')
+  const query = inputQuery || extractedQuery
+
+  if (inputTo && inputQuery) {
```

Comment:
⛏ ⛏ ⛏ this warning block should be up there, close to the error above, as it doesn't depend on anything else happening